examples/assets-and-custom-transfer-logic/assets/poi-approval.teal

```diff
@@ -0,0 +1,309 @@
+#pragma version 2
+// Creator Address
+// check if the app is being created
+// if so save creator
+int 0
+txn ApplicationID
+==
+bz not_creation
+txn NumAppArgs
+int 2
+==
+bz failed
+
+
+byte "Creator"
+txn Sender
+app_global_put
+byte "AssetID"
+txna ApplicationArgs 0
+btoi
+app_global_put
+byte "AssetLevel"
+txna ApplicationArgs 1
+btoi
+app_global_put
+int 1
+return
+
+
+not_creation:
+txn OnCompletion
+int NoOp
+==
+bnz handle_noop
+txn OnCompletion
+int OptIn
+==
+bnz handle_optin
+txn OnCompletion
+int CloseOut
+==
+bnz handle_closeout
+txn OnCompletion
+int UpdateApplication
+==
+bnz handle_updateapp
+txn OnCompletion
+int DeleteApplication
+==
+bnz handle_deleteapp
+// Unexpected OnCompletion value. Should be unreachable.
+b failed
+
+
+handle_noop:
+// Handle NoOp
+txn NumAppArgs
+int 1
+>=
+txn NumAppArgs
+int 2
+<=
+&&
+bz failed
+// should be one argument with
+// set-level, clear, or check-level
+// set-level and check-level have to be the creator
+// trade is the specific user trying to trade GLD
+txna ApplicationArgs 0
+byte "set-level"
+==
+bnz set-level
+
+txna ApplicationArgs 0
+byte "clear"
+==
+bnz clear
+txna ApplicationArgs 0
+byte "check-level"
+==
+bnz check-level
+err
+
+
+//only callable by creator
+//just one arg
+set-level:
+txn NumAppArgs
+int 2
+==
+byte "Creator"
+app_global_get
+txn Sender
+==
+&&
+global GroupSize
+int 1
+==
+&&
+bz failed
+int 1 //first account in accounts array
+byte "Accred-Level"
+txna ApplicationArgs 1
+btoi
+app_local_put
+int 1
+return
+
+
+clear:
+txn NumAppArgs
+int 1
+==
+byte "Creator"
+app_global_get
+txn Sender
+==
+&&
+global GroupSize
+int 1
+==
+&&
+bz failed
+int 1 //first account in accounts array
+byte "Accred-Level"
+app_local_del
+int 1
+return
+
+// check level should be called
+// with the following txes
+// tx 0 - check level smart contract call - signed by sender of asset
+// tx 1 - clawback transactions that moves the frozen asset from sender to receiver - signed by clawback-escrow
+// tx 2 - payment transaction from sender to clawback-escrow to pay for the fee of the clawback
+
+check-level:
+// check number and types of txes
+global GroupSize
+int 3
+==
+gtxn 0 TypeEnum
+int appl
+==
+&&
+gtxn 1 TypeEnum
+int axfer
+==
+&&
+gtxn 2 TypeEnum
+int pay
+==
+&&
+// Verify that this is index the stateful contract call
```

Comment:
what do you mean ?

Reply:
Means that for the fist tx group index is zero. When a txGroup is executed by default (without any stateless txn first, then groupIndex should be 0). I don't have 100% clarity regarding this.

Reply:
please modify comment, the sentence is not clear, maybe `the fist tx group index is zero. When a txGroup is executed by default (without any stateless txn first, then groupIndex should be 0)` or something else

Reply:
Ok, updating to your suggestion